lib/truffle/truffle/patching.rb

```diff
@@ -1,140 +1,68 @@
-require 'pathname'
-
 module Truffle::Patching
-  DIR     = Pathname(Truffle::Boot.ruby_home).join("lib/patches")
-  # Allowed operations are: :before, :after, :instead, :ignore
-  PATCHES = { "bundler"                                 => :before,
-              "bundler/version"                         => :after,
-              "bundler/cli/exec"                        => :after,
-              "bundler/compact_index_client/updater"    => :after,
-              "bundler/current_ruby"                    => :after,
-              "bundler/dependency"                      => :after,
-              "bundler/fetcher/downloader"              => :after,
-              "bundler/fetcher/compact_index"           => :after,
-              "bundler/ruby_version"                    => :after,
-              "bundler/source/rubygems"                 => :after,
-
-              "rubygems"                                => :before,
-              "rubygems/ext"                            => :after,
-              "rubygems/package"                        => :after,
-              "rubygems/remote_fetcher"                 => :after,
-              "rubygems/request"                        => :after,
-              "rubygems/request_set/gem_dependency_api" => :after,
-
-              "rspec/support"                           => :after,
-
-              "unf/normalizer_cruby"                    => :before,
-  }
-
-  PATCHES.each do |file, _|
-    unless DIR.join(file).sub_ext('.rb').exist?
-      raise "file #{file} is missing in #{DIR}"
+  def patches
+    @patches ||= begin
+      require 'pathname'
```

Comment:
with patching
```txt
time jt ruby --graal --trace --fg -J-verbose:gc -e 0
$ JAVACMD=/Users/pitr/Workspace/labs/graalvm-0.21/bin/java /Users/pitr/Workspace/labs/truffleruby-mx/bin/truffleruby -Xcore.load_path=/Users/pitr/Workspace/labs/truffleruby-mx/truffleruby/src/main/ruby -J-Dgraal.TruffleBackgroundCompilation=false -J-Dgraal.TraceTruffleCompilation=true -J-verbose:gc -e 0
[GC (Metadata GC Threshold)  27332K->2800K(125952K), 0.0037381 secs]
[Full GC (Metadata GC Threshold)  2800K->2651K(100864K), 0.0130903 secs]
[GC (Allocation Failure)  35931K->4291K(105472K), 0.0021554 secs]
[GC (Metadata GC Threshold)  11614K->4602K(119296K), 0.0020173 secs]
[Full GC (Metadata GC Threshold)  4602K->4107K(170496K), 0.0365541 secs]
[GC (Allocation Failure)  58891K->7709K(180736K), 0.0042113 secs]
[GC (Metadata GC Threshold)  14971K->7476K(180736K), 0.0033809 secs]
[Full GC (Metadata GC Threshold)  7476K->7035K(234496K), 0.0543271 secs]
[GC (Allocation Failure)  68987K->12288K(246272K), 0.0051951 secs]
[GC (Allocation Failure)  86528K->16541K(248832K), 0.0077263 secs]
[GC (Allocation Failure)  92317K->20007K(262144K), 0.0080738 secs]
[GC (Allocation Failure)  106535K->22884K(264192K), 0.0097185 secs]
[GC (Allocation Failure)  109412K->24990K(276992K), 0.0093854 secs]
[GC (Allocation Failure)  122270K->26214K(278016K), 0.0071188 secs]
[GC (Allocation Failure)  123494K->28418K(290816K), 0.0115019 secs]
[GC (Allocation Failure)  137474K->32108K(292864K), 0.0141744 secs]
[GC (Allocation Failure)  141164K->32628K(314368K), 0.0149787 secs]
[GC (Allocation Failure)  160628K->32792K(314880K), 0.0060489 secs]
[GC (Allocation Failure)  160792K->34410K(331776K), 0.0099865 secs]
[GC (Allocation Failure)  180330K->37412K(333824K), 0.0090357 secs]
[GC (Allocation Failure)  183332K->40962K(344576K), 0.0173394 secs]
[GC (Allocation Failure)  196098K->45982K(347136K), 0.0177265 secs]
[GC (Allocation Failure)  201118K->49517K(366080K), 0.0225265 secs]
[GC (Allocation Failure)  220013K->52516K(369152K), 0.0134794 secs]
        7.65 real        21.48 user         2.70 sys
```
without
```txt
$ JAVACMD=/Users/pitr/Workspace/labs/graalvm-0.21/bin/java /Users/pitr/Workspace/labs/truffleruby-mx/bin/truffleruby -Xcore.load_path=/Users/pitr/Workspace/labs/truffleruby-mx/truffleruby/src/main/ruby -J-Dgraal.TruffleBackgroundCompilation=false -J-Dgraal.TraceTruffleCompilation=true -J-verbose:gc -e 0
[GC (Metadata GC Threshold)  27431K->2851K(125952K), 0.0032604 secs]
[Full GC (Metadata GC Threshold)  2851K->2660K(101888K), 0.0131546 secs]
[GC (Allocation Failure)  35940K->4419K(101888K), 0.0020207 secs]
[GC (Metadata GC Threshold)  10705K->4531K(101888K), 0.0016151 secs]
[Full GC (Metadata GC Threshold)  4531K->3987K(151552K), 0.0239427 secs]
[GC (Allocation Failure)  37267K->6517K(155648K), 0.0029509 secs]
[GC (Metadata GC Threshold)  36237K->7644K(159232K), 0.0032296 secs]
[Full GC (Metadata GC Threshold)  7644K->7225K(218112K), 0.0543783 secs]
[GC (Allocation Failure)  48185K->11757K(223232K), 0.0037933 secs]
[GC (Allocation Failure)  58861K->15033K(224256K), 0.0055618 secs]
[GC (Allocation Failure)  62137K->17089K(232960K), 0.0056445 secs]
[GC (Allocation Failure)  70849K->19759K(233984K), 0.0074978 secs]
[GC (Allocation Failure)  73519K->21774K(241664K), 0.0124069 secs]
[GC (Allocation Failure)  81166K->22078K(243200K), 0.0065900 secs]
[GC (Allocation Failure)  81470K->22758K(254464K), 0.0083350 secs]
[GC (Allocation Failure)  91878K->24619K(255488K), 0.0080282 secs]
[GC (Allocation Failure)  93739K->25892K(268288K), 0.0112254 secs]
[GC (Allocation Failure)  106276K->27214K(270336K), 0.0091656 secs]
[GC (Allocation Failure)  107598K->29268K(287744K), 0.0135403 secs]
[GC (Allocation Failure)  125012K->29786K(289792K), 0.0108705 secs]
[GC (Allocation Failure)  125530K->29742K(304640K), 0.0160577 secs]
[GC (Allocation Failure)  138286K->31700K(306176K), 0.0107669 secs]
[GC (Allocation Failure)  140244K->35255K(318464K), 0.0167214 secs]
[GC (Allocation Failure)  154039K->38351K(321024K), 0.0160292 secs]
[GC (Allocation Failure)  157135K->40807K(339456K), 0.0209454 secs]
[GC (Allocation Failure)  175463K->45499K(341504K), 0.0191383 secs]
[GC (Allocation Failure)  180155K->52054K(364032K), 0.0240094 secs]
[GC (Allocation Failure)  206678K->52313K(367616K), 0.0157954 secs]
        7.74 real        21.61 user         2.73 sys
```
I don't think it has any impact. It is executed once and the pathname has just 478LoC, and does not require anything else.